controllers/dashboard.py

```diff
@@ -40,7 +40,7 @@ def index():
     course = db(db.courses.id == auth.user.course_id).select().first()
     assignments = db(db.assignments.course == course.id).select(db.assignments.ALL, orderby=db.assignments.name)
     logger.debug("getting chapters for {}".format(auth.user.course_name))
-    chapters = db(db.chapters.course_id == auth.user.course_name).select()
+    chapters = db(db.chapters.course_id == course.base_course).select()
```

Comment:
this is one of those places where the dynamic pages clash with the static pages... To preserve backward compatibility we may want to have a server side config variable

Reply:
Why run a server with static pages at all? I would contend that fewer options means more testing and more use of the code. Having lots of options / config vars means test case explosion, maintenance burden, and leads to cruft and code bloat. IMHO, supporting too many things means poor support.

So, I'd argue for **only** supporting dynamic routing if the server is in use to keep the code simple and clean. More config vars = more ways to screw up a server.

Reply:
I hear the complexity and testability arguments...  

The one big use case I was thinking about was for courses that build for people that are doing highly customized courses using Jenkins.  Whenever they update a Fork of one of the runestone books, Jenkins builds it and deploys it to runestone/static

I suppose that could be done as a dynamic build as well...

Reply:
(Lots of cheering for fewer options, fewer code paths, better test coverage...)